dot/state/block_test.go

```diff
@@ -195,6 +195,62 @@ func TestGetSlotForBlock(t *testing.T) {
 	require.Equal(t, expectedSlot, res)
 }
 
+func TestGetBlockHashesBySlot(t *testing.T) {
+	// create two block in the same slot and test if GetBlockHashesBySlot gets us
+	// both the blocks
+	bs := newTestBlockState(t, newTriesEmpty())
+	slot := uint64(77)
+
+	babeHeader := types.NewBabeDigest()
+	err := babeHeader.Set(*types.NewBabePrimaryPreDigest(0, slot, [32]byte{}, [64]byte{}))
+	require.NoError(t, err)
+	data, err := scale.Marshal(babeHeader)
+	require.NoError(t, err)
+	preDigest := types.NewBABEPreRuntimeDigest(data)
+
+	digest := types.NewDigest()
+	err = digest.Add(*preDigest)
+	require.NoError(t, err)
+	block := &types.Block{
+		Header: types.Header{
+			ParentHash: testGenesisHeader.Hash(),
+			Number:     1,
+			Digest:     digest,
+		},
+		Body: types.Body{},
+	}
+
+	err = bs.AddBlock(block)
+	require.NoError(t, err)
+
+	babeHeader2 := types.NewBabeDigest()
+	err = babeHeader2.Set(*types.NewBabePrimaryPreDigest(1, slot, [32]byte{}, [64]byte{}))
+	require.NoError(t, err)
+	data2, err := scale.Marshal(babeHeader2)
+	require.NoError(t, err)
+	preDigest2 := types.NewBABEPreRuntimeDigest(data2)
+
+	digest2 := types.NewDigest()
+	err = digest2.Add(*preDigest2)
+	require.NoError(t, err)
+	block2 := &types.Block{
+		Header: types.Header{
+			ParentHash: testGenesisHeader.Hash(),
+			Number:     1,
+			Digest:     digest2,
+		},
+		Body: types.Body{},
+	}
+	err = bs.AddBlock(block2)
+	require.NoError(t, err)
+
+	blocks, err := bs.GetBlockHashesBySlot(slot)
+	require.NoError(t, err)
+	require.Equal(t, len(blocks), 2)
+	require.Contains(t, blocks, block.Header.Hash())
+	require.Contains(t, blocks, block2.Header.Hash())
```

Comment:
simplifying with

```suggestion
	expectedBlockHashes := []common.Hash{block.Header.Hash(), block2.Header.Hash()}
	require.ElementsMatch(t, blocks, expectedBlockHashes)
```

also `require.Equal` has the expected as first arg, actual as 2nd arg, otherwise it's hard-to-debug test messages when it fails